assets/js/feature-tours/idea-hub-module.js

```diff
@@ -0,0 +1,102 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * External dependencies
+ */
+import { EVENTS } from 'react-joyride';
+
+/*
+ * Internal dependencies
+ */
+import { VIEW_CONTEXT_DASHBOARD } from '../googlesitekit/constants';
+import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+import { MODULES_IDEA_HUB } from '../modules/idea-hub/datastore/constants';
+
+const ideaHubModule = {
+	slug: 'ideaHubModule',
+	contexts: [ VIEW_CONTEXT_DASHBOARD ],
+	// TODO: change this if launch version for the feature changes.
+	version: '3.6.0',
+	checkRequirements: async ( registry ) => {
+		await registry.__experimentalResolveSelect( CORE_MODULES ).getModules();
+		const isIdeaHubModuleActive = registry.select( CORE_MODULES ).isModuleActive( 'idea-hub' );
+		const isIdeaHubModuleConnected = registry.select( CORE_MODULES ).isModuleConnected( 'idea-hub' );
+		const newIdeas = registry.select( MODULES_IDEA_HUB ).getNewIdeas() || [];
+
+		return ( isIdeaHubModuleActive && isIdeaHubModuleConnected && !! newIdeas.length );
+	},
+	steps: [
+		{
+			target: '.googlesitekit-idea-hub__title',
+			title: __( 'Get inspiration for new topics to write about!', 'google-site-kit' ),
+			content: __( 'These ideas are based on unanswered searches related to the content of your site. They are organised by topics and will refresh every N days.', 'google-site-kit' ),
+		},
+		{
+			target: '.googlesitekit-idea-hub__actions--create',
+			title: __( 'Start a draft', 'google-site-kit' ),
+			content: __( 'Found an interesting idea you want to write about? Create a draft! You can always get back to it later on the Posts page', 'google-site-kit' ),
+		},
+		{
+			target: '.googlesitekit-idea-hub__idea--single',
+			title: 'Save for later or dismiss',
+			content: "If you're not ready to create a draft about an idea just yet, add it to your 'Saved' list and revisit later. If you don’t like an idea, you can dismiss it from your list.",
+		},
+	],
+	gaEventCategory: 'idea_hub_module',
+	callback: ( data ) => {
+		/*
+		 * The third step of the feature tour involves the 'save' (pin) and
+		 * 'dismiss' buttons, both of which are hidden unless you hover over
+		 * them. As an enhancement, we will un-hide these buttons so they are
+		 * visible during the final step of the tour by adding a CSS class,
+		 * then remove the class once that step is finished.
+		 */
+		const unhideElementClass = 'googlesitekit-idea-hub__actions--unhide';
+
+		const pinElement = global.document.getElementsByClassName( 'googlesitekit-idea-hub__actions--pin' ).item( 0 );
+		const dismissElement = global.document.getElementsByClassName( 'googlesitekit-idea-hub__actions--delete' ).item( 0 );
+
+		if ( ! pinElement || ! dismissElement ) {
+			return;
+		}
+
+		const { type, index } = data;
```

Comment:
Let's move it to the beginning of the callback function and return early if `index` doesn't equal `2` or `type` is neither `STEP_BEFORE` nor `STEP_AFTER`.